spec/acceptance/namespaced_factories_spec.rb

```diff
@@ -0,0 +1,87 @@
+describe "Namespaced factories" do
+  before do
+    define_model("Category", name: :string)
+    define_model("Movies::Category", name: :string)
+  end
+
+  context "by default" do
+    before do
+      FactoryBot.define do
+        factory :category do
+          name { "Classics" }
+        end
+
+        with_namespace(:Movies) do
+          factory :category do
+            name { "Emmy Winners" }
+          end
+        end
```

Comment:
```suggestion
        factory :movies_category, class: "Movies::Category" do
          name { "Emmy Winners" }
        end
```

@zspencer Can you expand on what your suggestion gives us that this existing factory_bot syntax does not? 

@composerinteralia and I are working to understand what problem this is solving for your use case that's not covered by existing functionality. Thanks!

Reply:
Sure, given the following application structure:

`core` - A Rails Application that serves as the primary monolith
`product-a` - A Rails Engine that holds the Product A teams code
`product-b` A Rails Engine that holds the Product B teams code

We use FactoryBot across all three codebases; with the `core` code-base having a mix of end-to-end integration tests that want to use the Factories defined in `product-a` and `product-b`

In `core`, `product-a` and `product-b` all of them have `User` classes. In `core` it's simply `User`, but in `product-a` it is `ProductA::User` and in `product-b` it is `ProductB::User`.

When writing tests in the `product-a/spec/` suite, we want to use `FactoryBot.build(:user)` instead of `FactoryBot.build(:product_a_user)`. Same in the `product-b/spec/` suite.

However, in `core` we sometimes want to leverage traits defined in `product-a/spec/factories/product-a/users.rb`. To do so, we have shared the factories with the `core` code-base. _However_ because most of these factories were defined as `factory :user, class: "ProductA::User"` this results in a duplicate factory name definition.

To avoid having to go through and rename all the `FactoryBot.create(:user)` in `product-a/spec` to `FactoryBot.create(:product_a_user)` we sprouted the `with_namespace` method to wrap factories that are shared between `product-a`/`product-b` and `core`. 

Now, when in `product-b/spec/` we can call `FactoryBot.create(:user)` and it will use the factories in `product-b/spec/factories`, and in `core` we can call `FactoryBot.create(:user)` and it will use the top-level User, or `FactoryBot.create(:product_b_user)` and it will use the factory defined in `product-b/spec/factories/users.rb`

This is really only a useful feature in cases where a group has started to decompose a large monolith; so it may not be worth including in `FactoryBot` core.